connect-inject/container_init.go

```diff
@@ -298,31 +332,45 @@ cat <<EOF >/consul/connect-inject/service-defaults.hcl
 kind = "service-defaults"
 name = "{{ .ServiceName }}"
 protocol = "{{ .ServiceProtocol }}"
+{{- if .ConsulNamespace }}
+namespace = "{{ .ConsulNamespace }}"
+{{- end }}
 EOF
 {{- end }}
+
 {{- if .AuthMethod }}
 /bin/consul login -method="{{ .AuthMethod }}" \
   -bearer-token-file="/var/run/secrets/kubernetes.io/serviceaccount/token" \
   -token-sink-file="/consul/connect-inject/acl-token" \
+  {{- if .ConsulNamespace }}
+  -namespace="{{ .ConsulNamespace }}" \
```

Comment:
If mirroring is enabled, then the auth method is created in the default namespace. In that case, using `-namespace` flag results in this error:
```
Error logging in: Unexpected response code: 403 (rpc error making call: rpc error making call: ACL not found)
```

Reply:
Which is weird because I thought it was supposed to work globally when set in the default namespace...

Reply:
FWIW I've tried running it locally against a Consul agent on Kubernetes, and running the same command without `-namespace` works as expected (Consul puts the token in the token sink file) 🤷‍♀ .

Reply:
Adding the `-namespace` flag to this means that it's explicitly looking for an auth method in the given namespace. Because we've defined the auth method for mirroring in `default`, it doesn't find anything.

Removing the namespace flag means that the call falls back to running against the `default` namespace, which means that it works in this case because that's where the auth method is defined.

To be explicit, I'm going to set `-namespace="default"` for the mirroring case rather than relying on the fallback behavior. Just sorting out the tests around it now.

Reply:
Updated.